File: SDKs/shipengine.com/1.1.202006302006/python/src/sdk/models/operations/get_rate_by_id.py
import dataclasses
from typing import Optional
from ..shared import error_response_body as shared_error_response_body
from ..shared import get_rate_by_id_response_body as shared_get_rate_by_id_response_body


@dataclasses.dataclass
class GetRateByIDPathParams:
    rate_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'rate_id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetRateByIDRequest:
    path_params: GetRateByIDPathParams = dataclasses.field()
    

@dataclasses.dataclass
class GetRateByIDResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    error_response_body: Optional[shared_error_response_body.ErrorResponseBody] = dataclasses.field(default=None)
    get_rate_by_id_response_body: Optional[shared_get_rate_by_id_response_body.GetRateByIDResponseBody] = dataclasses.field(default=None)
    

File: SDKs/shipengine.com/1.1.202006302006/typescript/src/sdk/models/operations/getratebyid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetRateByIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=rate_id" })
  rateId: string;
}


export class GetRateByIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetRateByIdPathParams;
}


export class GetRateByIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  errorResponseBody?: shared.ErrorResponseBody;

  @SpeakeasyMetadata()
  getRateByIdResponseBody?: shared.GetRateByIdResponseBody;
}
